qa-test-app/src/main/java/com/mapbox/navigation/qa_test_app/view/RetainedActiveGuidanceFragment.kt

```diff
@@ -0,0 +1,60 @@
+package com.mapbox.navigation.qa_test_app.view
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.Fragment
+import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+import com.mapbox.navigation.core.lifecycle.MapboxNavigationApp
+import com.mapbox.navigation.qa_test_app.databinding.FragmentActiveGuidanceBinding
+import com.mapbox.navigation.qa_test_app.lifecycle.DropInActiveGuidance
+
+@OptIn(ExperimentalPreviewMapboxNavigationAPI::class)
+class RetainedActiveGuidanceFragment : Fragment() {
+
+    private lateinit var binding: FragmentActiveGuidanceBinding
+    private var dropInActiveGuidance: DropInActiveGuidance? = null
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        // Retaining the fragment is deprecated and not needed, but this example is to show
+        // how to build a non-leaking retained fragment.
+        retainInstance = true
+
+        // Because this component has an entirely different LifecycleOwner,
+        // attach it to the MapboxNavigationApp to ensure the DropInActiveGuidance will be
+        // able to connect to MapboxNavigation
+        MapboxNavigationApp.attach(this)
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        // Inflate the layout for this fragment
+        binding = FragmentActiveGuidanceBinding.inflate(inflater, container, false)
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        // Notice, the `DropInActiveGuidance` works the same if it is in a Fragment or an Activity.
+        dropInActiveGuidance = DropInActiveGuidance(
+            stopView = binding.stop,
+            maneuverView = binding.maneuverView,
+            tripProgressView = binding.tripProgressView
+        ).also { lifecycle.addObserver(it) }
+    }
+
+    override fun onDestroyView() {
+        // Null and remove. Note that this is to avoid memory leaks when retaining fragments,
```

Comment:
I believe this is also needed when not retaining fragments, because even then view can recreate within the same fragment instance. This happens when you remove the fragment or replace it and add this transaction to back stack. This can be observed in 1TAP, when navigating from map to settings, for example. 
So we always have to destroy `dropInActiveGuidance` to avoid leaking the view.

Reply:
You're right and thanks for calling out. updated the comment. Remembering all the reasons I stopped using fragments in favor of view-only architectures 😄

Reply:
Introducing a better pattern here will help us fix the issues downstream. I copy pasted this fragment from here
https://github.com/mapbox/mapbox-navigation-android-examples/blob/main/android-auto-app/src/main/java/com/mapbox/navigation/examples/androidauto/app/navigation/ActiveGuidanceFragment.kt

that's probably leaking, should be fixing this soon